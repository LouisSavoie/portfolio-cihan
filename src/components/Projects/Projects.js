import React from 'react'
import Project from '../Project/Project'

const Projects = ({ projects }) => {
  const displayProjects = () => {
    return projects?.map(project => {
      return <Project key={project.name} project={project} />
    })
  }

  return (
    <div data-testid="projects">
      {displayProjects()}
    </div>
  )
}

export default Projects
