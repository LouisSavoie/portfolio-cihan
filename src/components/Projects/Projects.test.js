import React from 'react'
import { render, screen, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom'
import renderer from 'react-test-renderer'
import Projects from './Projects'

afterEach(() => {
  cleanup()
})

test('should render Projects', () => {
  render(<Projects />)
  const ProjectsElement = screen.getByTestId('projects')
  expect(ProjectsElement).toBeInTheDocument()
})

test('matches snapshot', () => {
  const tree = renderer.create(<Projects />).toJSON()
  expect(tree).toMatchSnapshot()
})
