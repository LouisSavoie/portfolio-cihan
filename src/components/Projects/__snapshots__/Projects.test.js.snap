// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="container"
  data-testid="projects"
>
  <div
    className="justify-content-center row"
  >
    <h2>
      Projects
    </h2>
    <div
      className="card"
      data-testid="project"
      style={
        Object {
          "paddingBottom": ".5rem",
          "width": "24rem",
        }
      }
    >
      <img
        className="card-img-top"
        src="./img/bacon.png"
      />
      <div
        className="card-body"
      >
        <h3
          className="text-center card-title"
        >
          Bacon
        </h3>
        <div
          className="py-1 row"
        >
          <div
            className="col-4 py-1 col"
          >
            <div
              className="tech-item"
            >
              Short ribs
            </div>
          </div>
          <div
            className="col-4 py-1 col"
          >
            <div
              className="tech-item"
            >
              chuck
            </div>
          </div>
          <div
            className="col-4 py-1 col"
          >
            <div
              className="tech-item"
            >
              pastrami
            </div>
          </div>
          <div
            className="col-4 py-1 col"
          >
            <div
              className="tech-item"
            >
              corned beef
            </div>
          </div>
          <div
            className="col-4 py-1 col"
          >
            <div
              className="tech-item"
            >
              chicken
            </div>
          </div>
        </div>
        <p
          className="card-text"
        >
          Bacon ipsum dolor amet chislic ribeye drumstick buffalo jowl beef ribs cow bresaola pig brisket hamburger. Andouille picanha jerky cupim ground round filet mignon tenderloin sirloin venison ribeye turducken. Andouille tenderloin ground round pastrami rump pork chop, sausage buffalo porchetta kevin pancetta jerky hamburger. Shank pork beef, spare ribs tri-tip pork belly doner beef ribs strip steak meatball hamburger short loin jerky. Pork rump meatloaf frankfurter cow short loin drumstick venison tenderloin tail beef ribs. Ham alcatra chuck, boudin burgdoggen t-bone beef cow doner pastrami shoulder chicken drumstick capicola spare ribs.
        </p>
      </div>
      <div
        className="d-flex justify-content-center"
      />
    </div>
    <div
      className="card"
      data-testid="project"
      style={
        Object {
          "paddingBottom": ".5rem",
          "width": "24rem",
        }
      }
    >
      <img
        className="card-img-top"
        src="./img/ipsum.png"
      />
      <div
        className="card-body"
      >
        <h3
          className="text-center card-title"
        >
          Ipsum
        </h3>
        <div
          className="py-1 row"
        >
          <div
            className="col-4 py-1 col"
          >
            <div
              className="tech-item"
            >
              t-bone
            </div>
          </div>
          <div
            className="col-4 py-1 col"
          >
            <div
              className="tech-item"
            >
              turkey
            </div>
          </div>
          <div
            className="col-4 py-1 col"
          >
            <div
              className="tech-item"
            >
              tri-tip
            </div>
          </div>
        </div>
        <p
          className="card-text"
        >
          Tongue turkey salami, pig tenderloin chislic rump pancetta ham hock venison pork chop swine shoulder. Doner fatback leberkas pork shankle short ribs turkey prosciutto porchetta strip steak alcatra. Shank flank pig short ribs pork belly. Pork chop ball tip filet mignon, porchetta shankle kielbasa shoulder short ribs pork belly prosciutto landjaeger tail chislic buffalo fatback. Pork chop strip steak hamburger picanha. Swine turkey strip steak, short loin spare ribs drumstick ground round leberkas. Kevin tail filet mignon leberkas jerky bacon meatball boudin prosciutto tongue t-bone.
        </p>
      </div>
      <div
        className="d-flex justify-content-center"
      />
    </div>
    <div
      className="card"
      data-testid="project"
      style={
        Object {
          "paddingBottom": ".5rem",
          "width": "24rem",
        }
      }
    >
      <img
        className="card-img-top"
        src="./img/yum.png"
      />
      <div
        className="card-body"
      >
        <h3
          className="text-center card-title"
        >
          Yum
        </h3>
        <div
          className="py-1 row"
        >
          <div
            className="col-4 py-1 col"
          >
            <div
              className="tech-item"
            >
              rump
            </div>
          </div>
          <div
            className="col-4 py-1 col"
          >
            <div
              className="tech-item"
            >
              drumstick
            </div>
          </div>
          <div
            className="col-4 py-1 col"
          >
            <div
              className="tech-item"
            >
              brisket
            </div>
          </div>
          <div
            className="col-4 py-1 col"
          >
            <div
              className="tech-item"
            >
              sausage
            </div>
          </div>
          <div
            className="col-4 py-1 col"
          >
            <div
              className="tech-item"
            >
              ribeye
            </div>
          </div>
        </div>
        <p
          className="card-text"
        >
          Bacon beef ribs ball tip cupim rump pancetta, pork belly bresaola pork chop strip steak. Kevin short loin tail salami filet mignon ham hock, pig doner chislic. Picanha porchetta ground round brisket meatball. Leberkas jowl pork loin tri-tip ball tip hamburger. Beef meatloaf buffalo pastrami kielbasa leberkas ground round short ribs tri-tip pork belly venison bresaola salami.
        </p>
      </div>
      <div
        className="d-flex justify-content-center"
      />
    </div>
  </div>
</div>
`;
